tests/link/pytorch/test_basic.py

```diff
@@ -235,3 +235,20 @@ def test_arange():
         FunctionGraph([start, stop, step], [out]),
         [np.array(1), np.array(10), np.array(2)],
     )
+
+
+def test_eye():
+    N = scalar("N", dtype="int64")
+    M = scalar("M", dtype="int64")
+    k = scalar("k", dtype="int64")
+
+    out = eye(N, M, k, dtype="int16")
+
+    trange = range(1, 6)
+    for _N in trange:
+        for _M in trange:
+            for _k in list(range(_M + 2)) + [-x for x in range(1, _N + 2)]:
+                compare_pytorch_and_py(
+                    FunctionGraph([N, M, k], [out]),
+                    [np.array(_N), np.array(_M), np.array(_k)],
+                )
```

Comment:
In this case let's compile a single pytensor function and call repeatedly with the different inputs, instead of creating a function for every combination of test values. Something like:

```python
fn = pytensor.function([N, M, k], out, mode=torch_mode)  # Whatever mode is used internally in `compare_pytorch_and_py`
for _N in trange....:
  ...
  np.testing.assert_allclose(fn(_N, _M, _k), np.eye(_N, _M, _k))
```